### ISSUE TITLE
Incorrect Snake Case Auto Format

### ISSUE THREAD

**Issue by LcHicks** [DavidAnson/vscode-markdownlint#161]
I have found a bug which has forced me to set `"editor.formatOnSave": false,`:

My .md file has a blockquote with a snake_case file name and the underscores are randomly changed to asterisks.

Here is the original (path changed for confidentiality):

```Markdown
> folder/folder/folder/snake_case_file_name_that_is_long,1,0,  
> folder/folder/folder/snake_case_file_name_that_**has_some_bold_for_emphasis**,2,0,  
> folder/folder/folder/another_snake_case_file_name_that_is_long,3,0,  
> folder/folder/folder/yet_another_snake_case_file_name_that_is_long,4,0,
```

If I save, it changes to:

```Markdown
> folder/folder/folder/snake*case_file_name_that_is_long,1,0,  
> folder/folder/folder/snake_case_file_name_that***has_some_bold_for_emphasis**,2,0,  
> folder/folder/folder/another_snake_case_file_name_that_is_long,3,0,  
> folder/folder/folder/yet_another_snake_case_file_name_that_is_long,4,0,
```

An underscore on the first and third line were changed to asterisks.  
If I save a second time it changes it further:

```Markdown
> folder/folder/folder/snake\*case_file_name_that_is_long,1,0,  
> folder/folder/folder/snake_case_file_name_that**\*has_some_bold_for_emphasis**,2,0,  
> folder/folder/folder/another_snake_case_file_name_that_is_long,3,0,  
> folder/folder/folder/yet_another_snake_case_file_name_that_is_long,4,0,
```
I couldn't find a specific rule that demands this. Is there a way to disable this auto format rule?

**Comment by DavidAnson** [comment]
Your sample doesn't trigger any violations (see link below), but you're almost certainly hitting the "no spaces in emphasis" rule. You should be able to confirm by switching to "lint while typing" and looking at the text. This rule can get confused by scenarios that use a lot of underscores and asterisks within the text because it cannot rely on the heuristics that Markdown uses for valid input. If you disable this rule at the project file, or section-of-the-file level, it should not trigger and therefore should not get automatically fixed.

https://dlaa.me/markdownlint/#%25m%23%20Issue%20161%0A%0A%3E%20folder%2Ffolder%2Ffolder%2Fsnake_case_file_name_that_is_long%2C1%2C0%2C%20%20%0A%3E%20folder%2Ffolder%2Ffolder%2Fsnake_case_file_name_that_**has_some_bold_for_emphasis**%2C2%2C0%2C%20%20%0A%3E%20folder%2Ffolder%2Ffolder%2Fanother_snake_case_file_name_that_is_long%2C3%2C0%2C%20%20%0A%3E%20folder%2Ffolder%2Ffolder%2Fyet_another_snake_case_file_name_that_is_long%2C4%2C0%2C%0A

**Comment by LcHicks** [comment]
The linter does not give any warnings about rules that are not being met.

I tried disabling markdownlint in that section by adding `<!-- markdownlint-disable -->` before the section in question and the format on save still changed it. Interestingly, before changing `"editor.formatOnSave"` to `true` I tried disabling the markdownlint extension (and re-opening vscode) and it was still making this change when I saved.
